### PR TITLE
Fibo working

### PR DESCRIPTION
**Pull request by kuchichan**
Quite "regular" version.

### CODE REVIEW

File: module1/homework/fibonacci/fibonacci.hpp [hunk 1]
```diff
@@ -1,11 +1,21 @@
 #pragma once
+#include <cstddef>
 
 int fibonacci_iterative(int sequence) {
-    // TODO: Your implementation goes here
-    return 0;
+    int a = 0;
+    int b = 1;
+    int temp;
+    for (std::size_t i = 0; i < sequence; ++i) {
```
Comment (TalNovatzky) [inline module1/homework/fibonacci/fibonacci.hpp:8]: why do you need it std::size_t here instead of an integer?

Comment (kuchichan) [reply module1/homework/fibonacci/fibonacci.hpp:8]: TBH No serious reason :) I thought that size_t is more adequate during iteration than simple int
(from cpp reference)

> std::size_t is commonly used for array indexing and loop counting. Programs that use other types, such as unsigned int, for array indexing may fail on, e.g. 64-bit systems when the index exceeds UINT_MAX or if it relies on 32-bit modular arithmetic. 

Is that wrong in some way?

Comment (TalNovatzky) [reply module1/homework/fibonacci/fibonacci.hpp:8]: no, I was just curious, thank you for the answer.

Comment (TalNovatzky) [reply module1/homework/fibonacci/fibonacci.hpp:8]: now you learned me something ;)

**Comment by ziobron** [comment]
{ "requirement_ids": [1119, 1430], "points": [2, 6] }

**Pull request closed without merging**